src/utils/computeReClammInitAmounts.ts

```diff
@@ -0,0 +1,109 @@
+        initialTargetPrice,
+    );
+
+    const referenceToken = new Token(
+        chainId,
+        referenceAmountIn.address,
+        referenceAmountIn.decimals,
+        undefined, // symbol
+        undefined, // name
+        undefined, // wrapped
+        sortedTokens[givenTokenIndex]?.rate,
+    );
+    const computedToken = new Token(
+        chainId,
+        sortedTokens[calculatedTokenIndex].address,
+        sortedTokens[calculatedTokenIndex].decimals,
+        undefined, // symbol
+        undefined, // name
+        undefined, // wrapped
+        sortedTokens[calculatedTokenIndex]?.rate,
+    );
+
+    // Scale given amount to 18 decimals before doing math
+    const referenceTokenAmount = referenceToken.rate
+        ? TokenAmountWithRate.fromRawAmountWithRate(
+              referenceToken,
+              referenceAmountIn.rawAmount,
+              referenceToken.rate,
+          )
+        : TokenAmount.fromRawAmount(
+              referenceToken,
+              referenceAmountIn.rawAmount,
+          );
+
+    // Do math based on token order: https://github.com/balancer/reclamm/blob/8207b33c1ab76de3c42b015bab5210a8436376de/test/reClammPool.test.ts#L120-L128
+    const computedAmountScaled18 =
+        givenTokenIndex === 0
+            ? MathSol.mulDownFixed(referenceTokenAmount.scale18, proportion) // if given token is first in sort order, we multiply
+            : MathSol.divDownFixed(referenceTokenAmount.scale18, proportion); // if given token is second in sort order, we divide'
+
+    // Convert back to native token decimals of the calculated token (rounding down)
+    const computedAmountScaledRaw = computedToken.rate
+        ? // TODO: this is broken, how to fix so tokens scaled 18 are scaled down by rate?
+          TokenAmountWithRate.fromScale18AmountWithRate(
+              computedToken,
+              computedAmountScaled18,
+              computedToken.rate,
+          )
+        : TokenAmount.fromScale18Amount(computedToken, computedAmountScaled18);
+
+    const computedAmountIn: InputAmount = {
+        address: sortedTokens[calculatedTokenIndex].address,
+        rawAmount: computedAmountScaledRaw.amount,
+        decimals: sortedTokens[calculatedTokenIndex].decimals,
+    };
+
+    // // on chain sanity check for computedAmountScaled18
+    // const initialBalances = await client.readContract({
+    //     address: poolAddress,
+    //     abi: parseAbi([
+    //         'function computeInitialBalances(address referenceToken, uint256 referenceAmountIn) view returns (uint256[])',
+    //     ]),
+    //     functionName: 'computeInitialBalances',
+    //     args: [referenceToken.address, referenceTokenAmount.scale18],
+    // });
+    // console.log('on chain initialBalances:', initialBalances);
+
+    // TODO: sort token amounts within InitPool.buildCallWithPermit2 so this not needed
+    // Return amounts in consistent order based on token addresses
+    return [referenceAmountIn, computedAmountIn];
```

Comment:
If you're not doing this on this PR, please create an issue to keep track of it and delete the comment

Reply:
my bad i forgot to remove this comment after i realized assumption was incorrect as tokens still sorted by `InitPool.buildCall` during execution of `InitPool.buildCallWithPermit2`